examples/controls_example/public/react_controls/data_controls/search_control/get_search_control_factory.tsx

```diff
@@ -0,0 +1,227 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import React, { useEffect } from 'react';
+import deepEqual from 'react-fast-compare';
+import { BehaviorSubject, combineLatest, debounceTime, distinctUntilChanged } from 'rxjs';
+
+import { EuiFieldSearch, EuiFormRow, EuiRadioGroup } from '@elastic/eui';
+import { CoreStart } from '@kbn/core/public';
+import { DataViewsPublicPluginStart } from '@kbn/data-views-plugin/public';
+import { i18n } from '@kbn/i18n';
+import { useStateFromPublishingSubject } from '@kbn/presentation-publishing';
+
+import { initializeDataControl } from '../initialize_data_control';
+import { DataControlFactory } from '../types';
+import {
+  SearchControlApi,
+  SearchControlState,
+  SearchControlTechniques,
+  SEARCH_CONTROL_TYPE,
+} from './types';
+
+const allSearchOptions = [
+  {
+    id: 'match',
+    label: i18n.translate('controlsExamples.searchControl.searchTechnique.match', {
+      defaultMessage: 'Fuzzy match',
+    }),
+    'data-test-subj': 'searchControl__matchSearchOptionAdditionalSetting',
+  },
+  {
+    id: 'simple_query_string',
+    label: i18n.translate('controlsExamples.searchControl.searchTechnique.simpleQueryString', {
+      defaultMessage: 'Query string',
+    }),
+    'data-test-subj': 'optionsListControl__queryStringSearchOptionAdditionalSetting',
+  },
+];
+
+const DEFAULT_SEARCH_TECHNIQUE = 'match';
+
+export const getSearchEmbeddableFactory = ({
```

Comment:
nit: should we call this `getSearchControlFactory`? I think we already have a `searchEmbeddable`.

Reply:
Woops - yup, good call. Done in 0cd076e531cee1965afbb8736685b0182be61bee